src/main/java/de/aservo/confapi/commons/rest/AbstractApplicationLinksResourceImpl.java

```diff
@@ -31,6 +31,18 @@ public Response setApplicationLinks(
         return Response.ok(updatedLinksBean).build();
     }
 
+    @Override
+    public Response setApplicationLink(
+            final Long id,
+            final boolean ignoreSetupErrors,
+            ApplicationLinkBean linkBean) {
+        linkBean.setId(id);
+        final ApplicationLinkBean updatedLinkBean = applicationLinksService.setApplicationLink(
```

Comment:
Würde ID einfach an die Funktion als parameter übergeben statt linkBean.setID

Reply:
die ID wäre dann irgendwie doppelt gemoppeklt :-) da sie ja auch nochmal im bean drin steckt. ich dachte so wäre es eindeutiger, da bei der impl nur noch eine Stelle mit der ID übrig bleibt. da beim REST call immer die ID erzwungen wird im path wäre diese ID auch maßgeblich, unabhängig davon was vielleicht in der bean steckt....

Reply:
Im Request-Bean sollte die ID eigentlich nie gesetzt sein dürfen.

Und eigentlich sollten die Beans Immutable sein, d.h. alle Attribute final. Das ist etwas, das mir schon seit der Einführung von Lombok nicht gefällt, dass es nicht so ist. Aber wenn wir dort einen Builder generieren würden, wäre es umsetzbar. Ein solches Model Bean aus einem Request zu modifizieren ist meiner Ansicht nach "schlechter Stil", die ID einfach weiterzureichen tut dagegen nicht weh.